src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -952,6 +953,13 @@ public IAsyncResult BeginExecuteNonQuery(AsyncCallback callback, object stateObj
             }
         }
 
+        // This is in its own method to avoid always allocating the lambda in BeginExecuteNonQuery
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Why NoInlining?  Whether or not the JIT inlines this will have no affect on the IL the C# compiler generates that's responsible for when/where the closure is allocated.

Reply:
Removed.